jbpm-bpmn2/src/test/java/org/jbpm/bpmn2/DataTest.java

```diff
@@ -495,7 +615,109 @@ public void executeWorkItem(WorkItem workItem,
+                        Person person = (Person) processInstance.getVariable("instanceMetadata");
+                        assertNotNull(person);
+                        assertEquals("Napoleon", person.getName());
+                        assertEquals("Paris", person.getAddress().getCity());
+                        assertEquals("France", person.getAddress().getCountry());
+                        mgr.completeWorkItem(workItem.getId(), Collections.singletonMap("output", new Person(
+                                "Javierito", new Address("Sevilla", "Spain"))));
+                        assertEquals("Napoleon", person.getName());
+                        assertEquals("Sevilla", person.getAddress().getCity());
+                        assertEquals("Spain", person.getAddress().getCountry());
+                    }
+
+                });
+        ksession.startProcess("process", Collections.singletonMap("instanceMetadata", new Person("Napoleon",
+                        new Address("Paris", "France"))));
+    }
+
+    @Test
+    public void testDataOutputAssociationsWithPojoEmptyFrom() throws Exception {
+        KieBase kbase = createKnowledgeBaseWithoutDumper("BPMN2-DataOutputAssociations-Pojo-EmptyFrom.bpmn2");
+        ksession = createKnowledgeSession(kbase);
+        ksession.getWorkItemManager().registerWorkItemHandler("Human Task",
+                new WorkItemHandler() {
+
+                    public void abortWorkItem(WorkItem manager,
+                                              WorkItemManager mgr) {
+
+                    }
+
+                    public void executeWorkItem(WorkItem workItem,
+                                                WorkItemManager mgr) {
+                        WorkflowProcessInstance processInstance = (WorkflowProcessInstance) ksession.getProcessInstance(
+                                workItem.getProcessInstanceId());
+                        Person person = (Person) processInstance.getVariable("instanceMetadata");
+                        assertNotNull(person);
+                        assertEquals("Napoleon", person.getName());
+                        assertEquals("Paris", person.getAddress().getCity());
+                        assertEquals("France", person.getAddress().getCountry());
+                        mgr.completeWorkItem(workItem.getId(), Collections.singletonMap("instanceAddress", new Address(
+                                "Sevilla", "Spain")));
+                        assertEquals("Napoleon", person.getName());
+                        assertEquals("Sevilla", person.getAddress().getCity());
+                        assertEquals("Spain", person.getAddress().getCountry());
+                    }
+
+                });
+        ksession.startProcess("process", Collections.singletonMap("instanceMetadata", new Person("Napoleon",
+                new Address("Paris", "France"))));
+    }
+
+    @Test
+    public void testDataOutputAssociationsWithPojoList() throws Exception {
```

Comment:
I think it would be good to add a test to prove it works fine when using Maps as Anton's comment in the jira ticket. So we are covering all possible scenarios, wdyt?

Reply:
Done